### PR TITLE
feat: [UIE-8082] - DBaaS GA Database Resize refactoring

### PR DESCRIPTION
**Pull request by mpolotsk-akamai**
## Description 📝
 DBaaS GA: Database Resize refactoring

## Changes  🔄
List any change relevant to the reviewer.
- Refactoring Database Resize
   - The only real changes in this PR are `DatabaseResize.tsx`. Everything else is from the previous PR.

## Target release date 🗓️
11/12/24

### Prerequisites
(How to setup test environment)

### Verification steps
(How to verify changes)

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

**Comment by mjac0bs** [comment]
Hey @mpolotsk-akamai, this is a large PR with a diff of 1600+ lines, and it's introducing a few new components without test coverage.

Could we break this down into 2 PRs: Database Create updates in the first PR, and Database Resize refactor in the second?  This would lead to faster review and more isolated changes. I'm also not sure if lack of unit tests is intentional - are these changes going to be covered in Cypress flows in an upcoming ticket?

**Comment by mpolotsk-akamai** [comment]
> Hey @mpolotsk-akamai, this is a large PR with a diff of 1600+ lines, and it's introducing a few new components without test coverage.
> 
> Could we break this down into 2 PRs: Database Create updates in the first PR, and Database Resize refactor in the second? This would lead to faster review and more isolated changes. I'm also not sure if lack of unit tests is intentional - are these changes going to be covered in Cypress flows in an upcoming ticket?

Hey @mjac0bs, sure, I will break this down and will add unit tests. Thanks

**Comment by mpolotsk-akamai** [comment]
> > Hey @mpolotsk-akamai, this is a large PR with a diff of 1600+ lines, and it's introducing a few new components without test coverage.
> > Could we break this down into 2 PRs: Database Create updates in the first PR, and Database Resize refactor in the second? This would lead to faster review and more isolated changes. I'm also not sure if lack of unit tests is intentional - are these changes going to be covered in Cypress flows in an upcoming ticket?
> 
> Hey @mjac0bs, sure, I will break this down and will add unit tests. Thanks

Hey @mjac0bs, I created another PR https://github.com/linode/manager/pull/11193 for Database Create Summary Section, refactoring, and this existing PR #11180 is for Database Resize refactoring. 
However these PRs changes are connected, PR #11193 depends on PR #11180, because they have shared components.

**Review by mjac0bs — APPROVED** [review]
Thanks @mpolotsk-akamai - functionality looked good overall here.

Minor things:
- There's typecheck warnings in this PR that I did not see in the previous Create PR. You'll want to take a look at these imports. 
Screenshot 2024-11-04 at 11 35 40 AM

- I am still seeing the extra spacing before the number of nodes in the summary section, so I don't think that change got carried over when you were resolving conflicts. 
Screenshot 2024-11-04 at 11 44 04 AM

**Comment by mjac0bs** [comment]
Oh, there's some unit test flake still to fix. It passed the test-manager job in CI, but failed the current branch code coverage job. Locally, I can see it flake as well. 

image

File: packages/manager/src/features/Databases/DatabaseCreate/DatabaseCreate.tsx
Comment (cpathipa) [inline packages/manager/src/features/Databases/DatabaseCreate/DatabaseCreate.tsx:1]: Unused imports should be removed - FormControl, Radio, RadioGroup, RegionSelect, RegionHelperText,  and TextField

File: packages/manager/.changeset/pr-11193-added-1730373495513.md
[the remark this answers is not in the document]
Comment (mpolotsk-akamai) [reply packages/manager/.changeset/pr-11193-added-1730373495513.md:1]: Thank you for the review, @mjac0bs.
These errors were from the previous PR. I added changeset, updated branch with the latest changes from dev.

File: packages/manager/src/features/Databases/DatabaseCreate/DatabaseCreate.tsx
Comment (mpolotsk-akamai) [reply packages/manager/src/features/Databases/DatabaseCreate/DatabaseCreate.tsx:1]: Thank you for the review, @cpathipa .
These issues were from the previous PR. I updated this branch.

**Comment by mpolotsk-akamai** [comment]
@cpathipa , I'm working on it, thanks!

**Review by cpathipa — APPROVED** [review]
Thank you for addressing the feedback @mpolotsk-akamai

**Pull request merged**